src/main/java/org/folio/oaipmh/helpers/AbstractHelper.java

```diff
@@ -354,8 +355,11 @@ protected String getIdentifier(String identifierPrefix, String storageId) {
    * null if the result set is not partitioned.
    */
   protected ResumptionTokenType buildResumptionToken(Request request, JsonArray instances, Integer totalRecords) {
-    int newOffset = request.getOffset() + Integer.parseInt(RepositoryConfigurationUtil.getProperty
-      (request.getRequestId(), REPOSITORY_MAX_RECORDS_PER_RESPONSE)) + request.getInventoryOffsetShift();
+    int inventoryOffsetShift = request.getInventoryOffsetShift();
+    int maxRecordsPerResponse = Integer.parseInt(RepositoryConfigurationUtil.getProperty
+      (request.getRequestId(), REPOSITORY_MAX_RECORDS_PER_RESPONSE));
+    int newOffset = request.getOffset() + maxRecordsPerResponse + inventoryOffsetShift;
```

Comment:
Hi @obozhko-folio , I guess increment for newOffset value by **inventoryOffsetShift** should be done only once. Could please check that logic.